corehq/apps/sso/tests/test_async_handlers.py

```diff
@@ -341,6 +346,7 @@ def test_remove_object_raises_error_if_idp_is_editable_and_only_one_username(sel
         editable and there is only one UserExemptFromSingleSignOn relationship
         left for that IdentityProvider.
         """
+        self.addCleanup(self._cleanup_identity_provider)
```

Comment:
yes, because when the idp is created at the set up class level it is by default inactive/not editable. changing this on one test affects any tests that follow. This was the suggested approach from @millerdev

Reply:
From the [TestCase](https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.TestCase) docs: `[TestCase] Wraps the tests within two nested atomic() blocks: one for the whole class and one for each test`. In other words, `setUpClass` will get rolled back when the test suite itself is done, but any changes made to the data created by `setUpClass` will also get rolled back between tests.

Reply:
@mjriley i can assure you that in practice this is not true and the tests error out.

Reply:
Can you point me to anything I could try? I verified this locally.

Reply:
the default value of `self.idp.is_active` should be `False`. check the value of `self.idp.is_active` on a test that follows this test and you will see it's still stuck at `True`. Yes, the tests will pass, but the assumptions of `self.idp.is_active = False` at the start of each test would be incorrect. Luckily, no other tests that follow (right now) depend on this being true at the start, however, that does not make it correct to not add this cleanup in the event a test is added afterward that would depend on this assumption.

Reply:
Circling back: the tests work in line with the docs; what is going on here is that rolling database changes back does not affect class variables -- as soon as you call `self.idp.is_active`, the class variable will now be changed for all future tests, even though the database still believes `is_active` is False. There are a lot of ways to get around this without the need to call `_cleanup_identity_provider`. One way that had no visible effect on runtime was calling `self.idp.refresh_from_db()` in `setUp`.

`addCleanup` strikes me as a code smell -- I generally want resources to clean up after themselves, whether they're objects, locks, database connections, etc. I appreciate that the cleanup and setup are directly next to each other, but because this is abnormal for tests, I expect this to be a source of confusion as the code ages and we forget why we did what.  That said, it's not like calling `refresh_from_db()` will prevent someone doing the same thing with a different class variable; if you feel `addCleanup` is straight-forward, feel free to ignore this.

Reply:
thanks! `refresh_from_db()` was it. updating now

Reply:
there's also a few other places where this can get cleaned up in SSO tests. Since it's not relevant to this PR i will tackle that in a different PR.